pkg/rest/client/apiv1_client_opts.go

```diff
@@ -0,0 +1,36 @@
+package client
+
+import (
+	"net/http"
+	"time"
+)
+
+// ClientOptions is a struct that holds the options for the client
+type clientOptions struct {
+	transport http.RoundTripper
+	timeout   time.Duration
+}
+
+type ClientOption interface {
+	apply(*clientOptions)
+}
+
+// getDefaultClientOptions returns the default options for the client
+func getDefaultClientOptions() *clientOptions {
+	return &clientOptions{
+		timeout: 30 * time.Second,
+	}
+}
+
+type transportOption struct {
+	transport http.RoundTripper
+}
+
+func (t transportOption) apply(opts *clientOptions) {
+	opts.transport = t.transport
+}
+
+// WithTransport returns a function that sets the transport object
```

Comment:
Surprised the comment func mismatch didn't fail linting, I must have something misconfigured.

Reply:
Will fix the comment 👍